### ISSUE TITLE
Enable selection of font styles such as font names for Variables in Multi-Variable Text

### ISSUE THREAD

**Issue by hand-dot** [pdfme/pdfme#564]
### Is your feature request related to a problem? Please describe.

We have received several requests from many users wanting to specify font styles such as bold or italic.

Currently, it's not possible to specify font styles for specific parts of the text schema.

However, I thought that by enabling the selection of styles such as font names for Variables in Multi-Variable Text, users might be able to freely specify font styles.

### Describe the solution you'd like

Currently, Multi-Variable Text replaces placeholders with input and calls `text/pdfRender` as `parentPdfRender`.

I think this can be achieved by modifying or replacing `text/pdfRender` to retrieve and apply font names and other styles during rendering.

However, this modification would complicate the current simple PDF rendering process of Multi-Variable Text, so the implementation method needs to be carefully considered.

### Describe alternatives you've considered

none

**Comment by peteward** [comment]
Yes, this would need a considerable rewrite of the rendering both on-screen and in the PDF, so it is not a small task.

But it is a logical next step for the feature 👍

**Comment by Provello** [comment]
+1 For this feature - would be really great to support even very basic rich text

**Comment by onrdmr** [comment]
+1

**Comment by hand-dot** [comment]
ref: https://github.com/pdfme/pdfme/discussions/288#discussioncomment-11551608

**Comment by smiddleton93** [comment]
I would be willing to contribute to this feature as it's something I would use. In addition to the mentioned requested features I would also use variable font sizes and ordered/unordered lists

**Comment by smiddleton93** [comment]
https://github.com/user-attachments/assets/aa8de244-7d70-472c-bce5-ddb3e370458e

@hand-dot Have made a very simple start to a rich text plugin 
I have been using Tiptap as I am more familiar with it

I have also implemented text color, font sizes and lists in the generator, I just haven't got them implemented into the designer yet as I've had a bit of trouble getting my head around it

**Comment by peteward** [comment]
nice! 

How have you implemented the different font variations (italic and bold) in the generator?

**Comment by smiddleton93** [comment]
> nice!
> 
> How have you implemented the different font variations (italic and bold) in the generator?

SInce the default font is Roboto I'm currently just embedding the fonts Roboto-Bold, Roboto-Italic and Roboto-BoldItalic. The output of TipTap is html which is parsed into nodes of text/element tags. The nodes with the strong/em tags are flagged as bold/italic then mapped to the relevant font name. In the future I would probably have the schema allow a specific font name to specified for the different variations

**Comment by smiddleton93** [comment]
https://github.com/user-attachments/assets/ec5276f3-105e-48c1-9b38-24be3363460c 

<img width="810" alt="Image" src="https://github.com/user-attachments/assets/4f6360a3-ecdf-4279-a7fe-b65c91ec2389" />

Had trouble getting the functionality I needed from form inputs in the side bar so have been playing around with a floating toolbar. Still have some positioning issues to fix up in the generator but happy with progress

**Comment by hand-dot** [comment]
Hi @smiddleton93 !
That looks very good.
Please let me know if there's anything I can support you with.

**Comment by peteward** [comment]
Font selection will be interesting, you'll need additional bold and italic font dropdowns I guess?
I always thought that rendering will get tricky if people pick fonts that do not share the same size profile (especially height).

**Comment by bwl21** [comment]
This is a very important feature. Neverheless the question is how to implement this in the generator and how to represent it in Json. I think best would be Html. Based on this, we could use appropriate frontends

**Comment by hand-dot** [comment]
ref https://github.com/pdfme/pdfme/pull/1461

**Comment by superiority-claw-agent** [comment]
PDFweave already ships **inline-markdown** rich text in @pdfweave/schemas — see [`packages/schemas/src/text/richText.ts`](https://github.com/IDNTEQ/pdfweave/blob/main/packages/schemas/src/text/richText.ts). Bold / italic / inline code / line breaks via markdown syntax inside the text string. That covers the 70% case.

What we don't have: a Lexical-style WYSIWYG editor in the Designer. That's a substantial UI build that we haven't prioritised over the binding/anchors/stationery work. Tracking as v0.2+; would consider a contribution.
— filed by @superiority-claw-agent on behalf of the PDFweave maintainers (IDNTEQ). Context on the fork: pdfme/pdfme#1468.

**Comment by hand-dot** [comment]
Two existing PRs are now linked as incremental work toward these use cases: #1563 adds a selection-level inline-markdown formatting toolbar (including bold/italic), and #1557 carries per-run font sizes through rendering. Neither covers the complete rich-text request, particularly word-level highlighting, so these are related links rather than automatic closure links.

**Comment by hand-dot** [comment]
Maintainer triage update (current `main` checked at `7501849e`): status is `partially_done`.

Short rationale: current main has partial rich-text support via inline markdown: text schemas include `textFormat: 'plain' | 'inline-markdown'`, `fontVariants`, and `fontVariantFallback`, with rich-text UI/PDF rendering paths for markdown runs. However, the full requested workflow is not complete: there is no WYSIWYG rich-text editor on main, no selection-level toolbar on main, and no word-level highlight workflow. Normal editable text also hides the inline-markdown option unless the schema is read-only.

Evidence video: captured as a Cursor artifact at `/workspace/artifacts/evidence-564-rich-text.mp4`. The video shows inline-markdown rendering in the live Designer and notes the missing full toolbar/highlight authoring flow. To reproduce, use a read-only text schema with `textFormat: 'inline-markdown'` on current main and inspect the text panel/rendered output.

Suggested action: link #1563 as related toolbar work, but keep this issue open and update scope to distinguish existing inline-markdown support from remaining full rich-text/WYSIWYG/highlight requirements.

**Comment by hand-dot** [comment]
**What this recording shows:** Checking current rich-text surface: inline-markdown exists; full rich-text schema still missing.

Checked against `@pdfme/*` **6.1.13** (latest release) / `main` `7501849`.

![](https://github.com/user-attachments/assets/d3cce156-326a-4f09-8482-947b95cf055a)
